### ISSUE TITLE
Performance different between main program and object viewer

### ISSUE THREAD

**Issue by adfriz** [leezer3/OpenBVE#744]
When loading high poly count mesh or loading entire trainset(extension.cfg) to object viewer its gonna drop performance very significant.

I have test with my personal addon that have around 15000 opaque face and around 1000 tranparent face, 8x 1K texture and 2x 1K partially transparent texture its get around 30-50 fps without using mesh optimization, but when i play in main program the fps is significantly better around 50-65 fps(i use vsync because my monitor only 60hz).

I have test shinkansen E5 addon to object viewer and mainprogram but the result still the same, object viewer only got ~10fps when load the trainset and main program got 45-62 fps.

Why this happened ? Is this because main program only calculate the face it appear on screen but object viewer not or what???
Why not make object viewer as fast as the main program ??

**Comment by leezer3** [comment]
The main program performs various mesh optimisations, whereas Object Viewer does not.
There's also some other bits to do with visibility and face culling which are handled marginally differently.

**TLDR:**
Hand-coding objects (as per most BVE stuff) doesn't produce meshes which perform particularly well.
We can optimise a lot of this on-load (by converting quads to tris, squashing faces / materials etc.), but it means the wireframe view no longer matches what's written in the code, and this was considered desirable by some developers.

**Comment by ginga81** [comment]
Are you press 'R' key? If we press 'R' key, the renderer is change to GL3.0.

I think more serious issue what if the RouteViewer when we rotate suddenly increase or decrease the total alpha/opaque faces, the fps is incredibely slow.

At the future, if we can use the routeviewer's command at main program, we can more confortable.
If we can reload only the route data and new added object data, and if we can memory as the main program such as the same as the load in advance require to the routeviewer, I think maybe that we can more confortable.

At Bve5/6 ,when we prerss maybe SHIFT+F5, we can reload without reload object data and only the routedata.

**Comment by adfriz** [comment]
Leezer3
Wait you mean it performing all those optimization on load/ when select optimization mode to high ??

Ah thats answering why everytime i reload when using that high mode the loading time took very long (around 15-20 sec for high poly mesh)

Hmm i wonder can those optimization get separated in F10 window so it will have a separate option/checkbox to choose, did user want to convert quad to tris or just squashing faces or both when object reloaded.
If user choose both option it will perform the same like choosing high optimization mode.


Ginga81
I always use GL3.0 if i use old renderer the fps always get below 20.
